lib/livebook/apps/teams_app_spec.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Livebook.Apps.TeamsAppSpec do
+  # App spec for organization apps fetched from Livebook Teams.
+
+  @enforce_keys [:slug, :version, :archive_url, :derived_key]
+
+  # (apps) TODO: adjust archive download with new teams changes
+  defstruct [:slug, :version, :archive_url, :derived_key]
```

Comment:
@aleDsz this is a blueprint implementation that I haven't tested. Now that we no longer have the URL, we need to adjust the download logic accordingly.

There are just two other steps:

1. In the Provider implementation we need to build a list of app specs from app deployments. I also left a TODO.

2. In TeamsClient we need to ask the app manager to refresh the running apps, whenever we get new app deployment, redeployment or undeployment. In these cases you just need to call this:

    ```elixir
    defp manager_sync() do
      # Each node runs the teams client, but we only need to call sync once
      if Livebook.Apps.Manager.local?() do
        Livebook.Apps.Manager.sync_permanent_apps()
      end
    end
    ```

Reply:
Let's merge this PR and then we can revisit this and the deploy code. :)

Reply:
After this PR, we can update the implementation and integration tests accordingly with the changes. 🐈

Reply:
Done!